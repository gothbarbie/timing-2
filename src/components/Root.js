import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import defaultStore from '../store/configureStore'

const Root = ({ store, children }) => (
  <Provider store={store ? store : defaultStore}>{children}</Provider>
)

Root.propTypes = {
  store: PropTypes.shape({}),
  children: PropTypes.node,
}

export default Root
