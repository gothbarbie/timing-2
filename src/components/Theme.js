import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
