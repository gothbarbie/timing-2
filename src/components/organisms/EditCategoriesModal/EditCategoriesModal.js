import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'

import Button from 'generic/atoms/Button'
import Label from 'generic/atoms/Label'

import LabeledInput from 'generic/molecules/LabeledInput'
import Category from 'generic/molecules/Category'
import SearchCategories from 'generic/molecules/SearchCategories'

const FormSection = styled.section`
  display: grid;
  grid-template-columns: 4fr 1fr;
  grid-gap: 1rem;
  margin: 0 0 3rem 0;
  align-items: flex-end;
`

const RecentlyUsed = styled.section``

const CategoriesList = styled.div`
  margin-top: 1rem;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 0.5rem;
`

const AlignRight = styled.div`
  margin-top: 3rem;
  display: flex;
  justify-content: flex-end;
  width: 100%;
`

const EditCategoriesModal = ({ categories }) => (
  <section>
    <FormSection>
      <LabeledInput label="New" name="new-category" />
      <Button primary>Add</Button>
    </FormSection>

    <SearchCategories />

    <RecentlyUsed>
      <Label>Recently used</Label>
      <CategoriesList>
        {Object.entries(categories).map(([key, value]) => (
          <Category id={key} key={key} name={value.name} />
        ))}
      </CategoriesList>
    </RecentlyUsed>
    <AlignRight>
      <Button primary>Create</Button>
    </AlignRight>
  </section>
)

const mapStateToProps = ({ categories }) => ({
  categories,
})

export default connect(mapStateToProps)(EditCategoriesModal)
