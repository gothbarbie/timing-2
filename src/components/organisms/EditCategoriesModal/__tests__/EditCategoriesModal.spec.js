import React from 'react'
import { render } from 'enzyme'
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faSearch,
  faPencilAlt,
  faTimes,
} from '@fortawesome/free-solid-svg-icons'

import Root from 'components/Root'
import Theme from 'components/Theme'

import EditCategoriesModal from '../EditCategoriesModal'

library.add(faSearch, faPencilAlt, faTimes)

describe('components/organisms/EditCategoriesModal', () => {
  it('renders', () => {
    const component = render(
      <Root>
        <Theme>
          <EditCategoriesModal />
        </Theme>
      </Root>
    )

    expect(component).toMatchSnapshot()
  })
})
