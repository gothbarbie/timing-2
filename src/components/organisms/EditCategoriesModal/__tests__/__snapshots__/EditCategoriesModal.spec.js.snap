// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/organisms/EditCategoriesModal renders 1`] = `
<section>
  <section
    class="sc-jzJRlG bqpEsS"
  >
    <div
      class="sc-EHOje fpqoIO"
    >
      <label
        class="sc-bZQynM snbku"
        for="new-category"
      >
        New
      </label>
      <input
        class="sc-ifAKCX cYWJSo"
        name="new-category"
        type="text"
      />
    </div>
    <button
      class="sc-htpNat elmvaw"
    >
      Add
    </button>
  </section>
  <section
    class="sc-VigVT iJbgFv"
  >
    <div
      class="sc-EHOje fpqoIO"
    >
      <label
        class="sc-bZQynM snbku"
        for="search-category"
      >
        Search
      </label>
      <input
        class="sc-ifAKCX cYWJSo"
        name="search-category"
        type="search"
      />
    </div>
    <button
      class="sc-htpNat elmvaw"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 sc-gqjmRU desCyV"
        data-icon="search"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
  </section>
  <section
    class="sc-jTzLTM hRWMOj"
  >
    <div
      class="sc-fjdhpX WpPIX"
    >
      Search to get results
    </div>
  </section>
  <section
    class="sc-cSHVUG CAuLq"
  >
    <label
      class="sc-bxivhb jYacTM"
    >
      Recently used
    </label>
    <div
      class="sc-kAzzGY eOXwrh"
    >
      <div
        class="sc-htoDjs gRmJjR"
      >
        <div
          class="sc-iwsKbI IEitu"
        >
          test-1
        </div>
        <div
          class="sc-gZMcBi llkwiA"
        >
          <button
            class="sc-gzVnrw dwptPo"
            color="primary"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-pencil-alt fa-w-16 fa-2x "
              data-icon="pencil-alt"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            class="sc-gzVnrw dwptPo"
            color="primary"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times fa-w-11 fa-2x "
              data-icon="times"
              data-prefix="fas"
              role="img"
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="sc-htoDjs gRmJjR"
      >
        <div
          class="sc-iwsKbI IEitu"
        >
          test-2
        </div>
        <div
          class="sc-gZMcBi llkwiA"
        >
          <button
            class="sc-gzVnrw dwptPo"
            color="primary"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-pencil-alt fa-w-16 fa-2x "
              data-icon="pencil-alt"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            class="sc-gzVnrw dwptPo"
            color="primary"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times fa-w-11 fa-2x "
              data-icon="times"
              data-prefix="fas"
              role="img"
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </section>
  <div
    class="sc-chPdSV gZWPpb"
  >
    <button
      class="sc-htpNat elmvaw"
    >
      Create
    </button>
  </div>
</section>
`;
